### PR TITLE
feat(eventing): add proto serializable timestamp

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Added serve-proto crate for serialisable timestamp. Remove dependency on prost-wkt-types::Timestamp.

### CODE REVIEW

File: serde-proto/src/timestamp.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+use serde::{de::Visitor, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::common_types::Timestamp;
+
+impl TryFrom<Timestamp> for DateTime<Utc> {
+    type Error = &'static str;
+
+    fn try_from(value: Timestamp) -> Result<Self, Self::Error> {
+        let dt = DateTime::<Utc>::from_timestamp(value.seconds, value.nanos)
+            .expect("The datetime is not valid or falls outside the valid range");
```
Comment (tiagolobocastro) [inline serde-proto/src/timestamp.rs]: This should return error rather than expect?

File: serde-proto/protobuf/v1/common_types.proto [hunk 2]
```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+
+package v1.common_types;
+
+
+message Timestamp {
```
Comment (tiagolobocastro) [inline serde-proto/protobuf/v1/common_types.proto]: We don't need to define our own, we just add
import "google/protobuf/timestamp.proto";
and then use this `Timestamp` ?

File: serde-proto/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "serde-proto"
```
Comment (tiagolobocastro) [inline serde-proto/Cargo.toml]: it's not just serde but also date-time so maybe suggest renaming to something else?
Not a good suggestion, but example could do similar to the wtk crate:
wkt-proto

File: serde-proto/protobuf/v1/common_types.proto [hunk 4]
```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+
+package v1.common_types;
```
Comment (tiagolobocastro) [inline serde-proto/protobuf/v1/common_types.proto]: Separate common into specific protos if we need more kinds in the future
```suggestion
package v1.pb_time;
```

File: serde-proto/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+mod timestamp;
+
+/// Event module for the autogenerated event code.
+pub mod common_types {
+    #![allow(clippy::derive_partial_eq_without_eq)]
+    #![allow(clippy::large_enum_variant)]
+    tonic::include_proto!("v1.common_types");
+}
```
Comment (tiagolobocastro) [inline serde-proto/src/lib.rs]: Control what we export?
```suggestion
}
pub use common_types::Timestamp;
```

File: serde-proto/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+mod timestamp;
+
+/// Event module for the autogenerated event code.
+pub mod common_types {
```
Comment (tiagolobocastro) [inline serde-proto/src/lib.rs]:
```suggestion
mod common_types {
```

File: serde-proto/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+version = "0.1.0"
+edition = "2021"
+
+
+[build-dependencies]
+tonic-build = "0.8.4"
+
+[dependencies]
+tracing = "0.1.37"
```
Comment (tiagolobocastro) [inline serde-proto/Cargo.toml]: Is tracing an tonic/tonic-build needed here?

File: serde-proto/protobuf/v1/common_types.proto (hunk 2 above)
Comment (datacore-vvarakantham) [reply serde-proto/protobuf/v1/common_types.proto]: Yes. We need to also add
extern_path(".google.protobuf.Timestamp", "::serde_extend::Timestamp")
 while build the protobuf to use this timestamp

File: serde-proto/Cargo.toml (hunk 3 above)
Comment (datacore-vvarakantham) [reply serde-proto/Cargo.toml]: Do  I use 
prost-utils
prost-helpers
Prost-common-utils

Similar to these?

Comment (tiagolobocastro) [reply serde-proto/Cargo.toml]: Not keen on the last one but the others sound better.
I'll add a few more to choose from
prost-boost
prost-extend

File: serde-proto/src/timestamp.rs (hunk 1 above)
Comment (datacore-vvarakantham) [reply serde-proto/src/timestamp.rs]: Yes, returning the error now.

File: serde-proto/Cargo.toml (hunk 3 above)
Comment (datacore-vvarakantham) [reply serde-proto/Cargo.toml]: Using prost-extend.

File: serde-proto/protobuf/v1/common_types.proto (hunk 4 above)
Comment (datacore-vvarakantham) [reply serde-proto/protobuf/v1/common_types.proto]: Done.

File: serde-proto/Cargo.toml (hunk 7 above)
Comment (datacore-vvarakantham) [reply serde-proto/Cargo.toml]: Tracing is not needed. Removed it now. The others are needed.

**Review by tiagolobocastro — APPROVED** [review]

**Review by chriswldenyer — APPROVED** [review]
Looks believable.

**Pull request merged**